### PR TITLE
✨ [extension] filter outdated events

### PR DESCRIPTION
**Pull request by BenoitZugmeyer**
## Motivation

Showing each View update is confusing as it can be seen as separate view events.

## Changes

* Add a checkbox to show only the last view update
* Make sure View updates are collocated when many updates are shown

### Before:

Screenshot 2023-08-24 at 12 05 15

### After:

Outdated view updates are hidden

Screenshot 2023-08-24 at 12 06 23

View updates are collocated

Screenshot 2023-08-24 at 12 06 28

## Testing

- [x] Local
- [ ] Staging
- [ ] Unit
- [ ] End to end

---

I have gone over the [contributing](https://github.com/DataDog/browser-sdk/blob/main/CONTRIBUTING.md) documentation.

### CODE REVIEW

**Comment by BenoitZugmeyer** [comment]
Current dependencies on/for this PR:
* main
  * **PR #2369** <a href="https://app.graphite.dev/github/pr/DataDog/browser-sdk/2369" target="_blank">Graphite</a>  👈
    * **PR #2370** <a href="https://app.graphite.dev/github/pr/DataDog/browser-sdk/2370" target="_blank">Graphite</a> 
      * **PR #2371** <a href="https://app.graphite.dev/github/pr/DataDog/browser-sdk/2371" target="_blank">Graphite</a> 

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/DataDog/browser-sdk/2369?utm_source=stack-comment).

**Review by amortemousque — APPROVED** [review]

File: developer-extension/src/panel/hooks/useEvents/eventCollection.ts [hunk 1]
```diff
@@ -28,6 +26,17 @@ export function startEventCollection(strategy: EventCollectionStrategy, onEvents
...
 
+function compareEvents(a: SdkEvent, b: SdkEvent) {
+  // Sort events chronologically
+  if (a.date !== b.date) {
+    return b.date - a.date
+  }
+
+  // If two events have the same date, make sure to display View events last. This ensures that View
+  // updates are collocated in the list (no other event are present between two updates)
+  return (isRumViewEvent(a) as any) - (isRumViewEvent(b) as any)
```
Comment (bcaudan) [inline developer-extension/src/panel/hooks/useEvents/eventCollection.ts:37]: **❓ question:** in which case could we have two view updates with the same date?

File: developer-extension/src/panel/components/tabs/eventsTab.tsx [hunk 2]
```diff
@@ -43,7 +42,13 @@ export function EventTab({ events, filters, onFiltered, clear }: EventTabProps)
           >
             <Chip value="rum">RUM</Chip>
             <Chip value="logs">Logs</Chip>
+            <Chip value="telemetry">Telemetry</Chip>
           </Chip.Group>
+          <Checkbox
+            label="Show only the latest View event"
+            checked={!filters.outdatedVersions}
+            onChange={(e) => onFiltered({ ...filters, outdatedVersions: !e.target.checked })}
+          />
```
Comment (bcaudan) [inline developer-extension/src/panel/components/tabs/eventsTab.tsx:51]: **💬 suggestion:** ‏it seems to be the case with existing checkboxes but could we have a pointer cursor when hovering them (like the other inputs)?

File: developer-extension/src/panel/hooks/useEvents/eventCollection.ts (hunk 1 above)
Comment (BenoitZugmeyer) [reply developer-extension/src/panel/hooks/useEvents/eventCollection.ts:37]: We are talking about the event date here. So all View updates related to the same View have the same date.

Comment (BenoitZugmeyer) [reply developer-extension/src/panel/hooks/useEvents/eventCollection.ts:37]: I'll add an example in the comment to make it clearer.

File: developer-extension/src/panel/components/tabs/eventsTab.tsx (hunk 2 above)
Comment (BenoitZugmeyer) [reply developer-extension/src/panel/components/tabs/eventsTab.tsx:51]: Done! (luckily `mantine` supports it 🎉 )

**Review by bcaudan — APPROVED** [review]

**Pull request merged**